topics/go/algorithms/data/heap/heap.go

```diff
@@ -0,0 +1,140 @@
+// Package heap implements a heap interface, and add ability to store values of type Data in the minimum Heap.
+package heap
+
+import (
+	"container/heap"
+	"errors"
+)
+
+// ErrEmptyHeap returned if heap is empty
+var ErrEmptyHeap = errors.New("heap is empty")
+
+// Data represents what is being stored on the heap.
+type Data struct {
+	Value int
+	Index int
+}
+
+// Heap represents a list of data.
+type Heap struct {
+	size     int
+	capacity int
+	dataHeap *dataHeap
+}
+
+// New returns a new Heap with given capacity in which we will store our data.
+func New(cap int) (*Heap, error) {
+
+	if cap <= 0 {
+		return nil, errors.New("invalid capacity")
+	}
+
+	h := Heap{
+		size:     0,
+		capacity: cap,
+		dataHeap: &dataHeap{},
+	}
+
+	return &h, nil
+}
+
+// Store is storing the element of type Data in the heap.
+func (h *Heap) Store(data *Data) error {
+
+	// If we try to add data to full Heap, we will return an error.
+	if h.size >= h.capacity {
+		return errors.New("getting out of heap capacity")
```

Comment:
read this error message.  should we remove the word `getting` and just keep  `out of heap capacity`

Reply:
I agree with you.